tracer/src/Datadog.Trace/Ci/GitInfo.cs

```diff
@@ -92,7 +92,20 @@ private GitInfo()
         /// <returns>Git info</returns>
         public static GitInfo GetFrom(string folder)
         {
-            return GetFrom(new DirectoryInfo(folder));
+            // Try to load git metadata from the folder
+            if (TryGetFrom(new DirectoryInfo(folder), out var gitInfo))
+            {
+                return gitInfo;
+            }
+
+            // If not let's try to find the .git folder in a parent folder
+            if (TryGetFrom(GetParentGitFolder(folder), out var pFolderGitInfo))
+            {
+                return pFolderGitInfo;
+            }
+
+            // Return the partial gitInfo instance with the initial source root
+            return gitInfo;
```

Comment:
Hmmm, that's pretty unusual pattern for a `TryGet*` method 🤔 I would assume that if `TryGetFrom` returns `false`, the output variable is _not_ valid to use. Maybe we should rename it to `TryPopulateFrom` and pass in the `gitInfo` as a variable instead? 🤔 Doesn't _really_ matter, this just looks "wrong" to me 🤷‍♂️

Reply:
fair, let me change that

Reply:
done in [ce8fd1a](https://github.com/DataDog/dd-trace-dotnet/pull/5425/commits/ce8fd1a57f8178e5c40412e30a39dbc397534420)